### DISCUSSION TITLE
Can I create a plugin offline using js, directly writing in plugin folder

### DISCUSSION THREAD

**Discussion by shutaozhenzhen** [Eugeny/tabby#9781] [Q&A]
I'm work in a offline env, so I can not get plugin online. Also, I want to use tabby plugin function to create some simple plugin, but for the offline env, I can not using npm.

I'm thinking, can I just write js in plugin folder and tabby can detecte and load it?

**Comment by Eugeny** [comment]
Yes, see https://github.com/Eugeny/tabby/blob/master/HACKING.md#plugins

**Reply by shutaozhenzhen** [reply]
I don't think this is what I want.

The code in this guide is writen in ts, and need use webpack to compile it

**Reply by Eugeny** [reply]
You can just write in JS if you want.
